src/librustc_mir/dataflow/impls/borrows.rs

```diff
@@ -236,9 +285,65 @@ impl<'a, 'gcx, 'tcx> BitDenotation for Borrows<'a, 'gcx, 'tcx> {
         self.kill_loans_out_of_scope_at_location(sets, location);
     }
 
-    fn terminator_effect(&self,
-                         sets: &mut BlockSets<BorrowIndex>,
-                         location: Location) {
+    fn statement_effect_on_activations(&self,
```

Comment:
It's mildly surprising that this function **also** applies the statement effect on reservations. Perhaps rename to `statement_effect_on_activations_and_reservations`?

Reply:
(Alternatively move that call into the caller, but I think it's maybe nice to have them grouped together here? I guess I could go either way, just want the name to be clear on what is affected.)

Reply:
yeah I considered having that call in the caller, but I thought at some point there might be a case where the activation effect would want to observe state from both before and after the reservation effect. (This turned out not to be the case in the code so far, but I didn't want to build in an assumption that things would stay that way.)

Anyway the idea in my head is more that the ActiveBorrows bitvector holds *both* activations and reservations state. So maybe the problem is the naming here, I need some way to succinctly say "the combined reservation and activation state."

`PhasedBorrows`?

Reply:
I think I'm going to change this so that there's a single method and we pass a `is_activations: bool` parameter representing the context.